crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -2883,61 +2931,208 @@ impl Analyzer<'_, '_> {
                         }
                         .into());
                     }
-
+                };
+
+                return Err(ErrorKind::NotSatisfyConstraint {
+                    span: to.type_args.params[0].span(),
+                    left: box Type::Keyword(KeywordType {
+                        kind: TsKeywordTypeKind::TsStringKeyword,
+                        span: to.span(),
+                        metadata: Default::default(),
+                        tracker: Default::default(),
+                    }),
+                    right: box Type::Keyword(KeywordType {
+                        kind: ty,
+                        span: r.span(),
+                        metadata: Default::default(),
+                        tracker: Default::default(),
+                    }),
+                }
+                .into());
+            }
+            Type::Union(union) => {
```

Comment:
Can you use another name instead of `union`? GitHub highlighter treats this as a keyword.